classes/acrn-sblimage.bbclass

```diff
@@ -0,0 +1,87 @@
+inherit python3native
+DEPENDS += "slimboot-tools-native"
+
+MB_DEPENDENCY    ?= ""
+MB_MC_DEPENDENCY ?= ""
+MB_ACRN_BINARY   ?= "${DEPLOY_DIR_IMAGE}/acrn.32.out"
+MB_ACRN_CMDLINE  ?= ""
+MB_ACRN_MODULES  ?= "${TOPDIR}/conf/linux.txt;${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGETYPE}"
+BASE_SBLIMAGE    ?= "sbl_os"
+SBLIMAGE_NAME    ?= "${BASE_SBLIMAGE}"
+PREGENERATED_SIGNING_KEY_SLIMBOOT_KEY_SHA256 ?= "${TOPDIR}/cert/TestSigningPrivateKey.pem"
+
+python do_acrn_sblimage() {
+    import re
+    import os
+
+    mbAcrnBinaryDeployDir = d.getVar('MB_ACRN_BINARY').lstrip().rstrip()
+    if not os.path.isfile(mbAcrnBinaryDeployDir):
+        bb.fatal("acrn %s not found!" % mbAcrnBinaryDeployDir)
+
+    mbAcrnCmdlineDeployDir = d.getVar('MB_ACRN_CMDLINE').lstrip().rstrip()
+    if mbAcrnCmdlineDeployDir == "":
+        hv_cmdline = d.getVar('WORKDIR') + "/hv_cmdline"
+        bb.debug(1, "hv_cmdline: %s" % (hv_cmdline))
+        os.system("echo 'uart=mmio@0xfe042000' > %s" % (hv_cmdline))
+    else:
+        hv_cmdline = mbAcrnCmdlineDeployDir
+
+    mbAcrnBinaryDeployDirPairList = re.split(" +", d.getVar('MB_ACRN_MODULES').lstrip().rstrip())
+
+    genContainerPath = "%s/%s/slimboot/Tools/GenContainer.py" % (d.getVar('STAGING_DIR_NATIVE'), d.getVar('libexecdir'))
+
+    genContainerCmd = "%s" % d.getVar('PYTHON')
+    genContainerCmd = genContainerCmd + " %s create" % (genContainerPath)
+    genContainerCmd = genContainerCmd + " -cl"
+    genContainerCmd = genContainerCmd + " CMDL:%s" % (hv_cmdline)
+    genContainerCmd = genContainerCmd + " ACRN:%s" % (mbAcrnBinaryDeployDir)
+    counter = 0
+    for mbAcrnBinaryDeployDirPair in mbAcrnBinaryDeployDirPairList:
+        mbAcrnBinaryDeployDirPairMods = re.split(";", mbAcrnBinaryDeployDirPair.strip())
+        bb.debug(1, "module[%s]: %s" % (counter,   mbAcrnBinaryDeployDirPairMods[0]))
+        bb.debug(1, "module[%s]: %s" % (counter+1, mbAcrnBinaryDeployDirPairMods[1]))
+
+        if not os.path.isfile(mbAcrnBinaryDeployDirPairMods[0]):
+            bb.fatal("module %s not found!" % mbAcrnBinaryDeployDirPairMods[0])
+
+        if not os.path.isfile(mbAcrnBinaryDeployDirPairMods[1]):
+            bb.fatal("module %s not found!" % mbAcrnBinaryDeployDirPairMods[1])
+
+        genContainerCmd = genContainerCmd + " MOD%s:%s MOD%s:%s" % (counter, mbAcrnBinaryDeployDirPairMods[0], counter+1, mbAcrnBinaryDeployDirPairMods[1])
+        counter = counter + 2
+    genContainerCmd = genContainerCmd + " -o %s/%s" % (d.getVar('WORKDIR'), d.getVar('SBLIMAGE_NAME'))
+    genContainerCmd = genContainerCmd + " -k %s" % (d.getVar('PREGENERATED_SIGNING_KEY_SLIMBOOT_KEY_SHA256'))
+    genContainerCmd = genContainerCmd + " -t MULTIBOOT"
+
+    bb.debug(1, "genContainerCmd: %s" % (genContainerCmd))
+    os.system("%s" % (genContainerCmd))
+    os.system("install -d %s/boot; install -m 644 %s/%s %s/boot" % (d.getVar('IMAGE_ROOTFS'), d.getVar('WORKDIR'), d.getVar('SBLIMAGE_NAME'), d.getVar('IMAGE_ROOTFS')))
```

Comment:
Sorry I had not noticed this in the first revision. I think these two calls should be replaced by subprocess.